uw-frame-components/portal/messages/controllers.js

```diff
@@ -0,0 +1,590 @@
+'use strict';
+
+define(['angular'], function(angular) {
+  return angular.module('portal.messages.controllers', [])
+
+    .controller('MessagesController', ['$q', '$log', '$scope', '$rootScope',
+      '$location', '$localStorage', '$sessionStorage', '$filter', '$mdDialog',
+      'MESSAGES', 'SERVICE_LOC', 'miscService', 'messagesService',
+      function($q, $log, $scope, $rootScope, $location, $localStorage,
+               $sessionStorage, $filter, $mdDialog, MESSAGES, SERVICE_LOC,
+               miscService, messagesService) {
+        // //////////////////
+        // Local variables //
+        // //////////////////
+        var allMessages = [];
+        var promiseFilteredMessages = {};
+
+        // ////////////////
+        // Local methods //
+        // ////////////////
+        /**
+         * Get all messages, then pass result on for filtering
+         */
+        var getMessages = function() {
+          messagesService.getAllMessages()
+            .then(function(result) {
+              // Ensure messages exist and check for group filtering
+              if (result.messages && result.messages.length > 0) {
+                allMessages = result.messages;
+              }
+              filterMessages();
+              return allMessages;
+            })
+            .catch(function(error) {
+              $log.warn('Problem getting all messages for messages controller');
+              return error;
+            });
+        };
+
+        /**
+         * Determine whether or not messages need to be filtered
+         * by group and data, then execute the relevant promises
+         */
+        var filterMessages = function() {
+          // Check for group filtering on app level and beta settings level
+          if (MESSAGES.groupFiltering
```

Comment:
There's a setting on the secret settings page to toggle.  I used it last week to verify a mascot announcement, since I wasn't in the group.  I think we can get rid of it for now and deal with it.  We'll see if it's important enough to make a comeback.